extra_foam/gui/windows/pump_probe_w.py

```diff
@@ -138,8 +148,61 @@ def initUI(self):
 
     def initConnections(self):
         """Override."""
-        pass
+        self._ctrl_widget.save_btn.clicked.connect(self.saveToFile)
 
     def closeEvent(self, QCloseEvent):
         self._ctrl_widget.resetAnalysisType()
         super().closeEvent(QCloseEvent)
+
+    def saveToFile(self):
+        # Get the current data
+        if not len(self._queue):
+            return
```

Comment:
It would be better if the button itself was disabled when there is no data to save, rather than silently exiting.